### ISSUE TITLE
Invoke-WebRequest change of TypeName of Headers.Values

### ISSUE THREAD

**Issue by CarloToso** [PowerShell/PowerShell#18884]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

The TypeName of Headers.Values is `System.String` in PS 5 and changes to `System.String[]` in PS 7. Is this the desired behaviour?

### Expected behavior

```console
(Invoke-WebRequest "http://postman-echo.com/get").Headers.Values | Get-Member

TypeName: System.String
```

### Actual behavior

```console
(Invoke-WebRequest "http://postman-echo.com/get").Headers.Values | Get-Member

TypeName: System.String[]
```

### Environment data

```powershell
PS C:\Users\carlo> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.1
PSEdition                      Core
GitCommitId                    7.3.1
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by mklement0** [comment]
I can't speak to why this change happened, but just to provide the relevant links:

The `.Headers` property type changed from:

`System.Collections.Generic.Dictionary<string,string>` in [Windows PowerShell](https://learn.microsoft.com/en-us/dotnet/api/microsoft.powershell.commands.webresponseobject.headers?view=powershellsdk-1.1.0#microsoft-powershell-commands-webresponseobject-headers)

to:

`System.Collections.Generic.Dictionary<string,System.Collections.Generic.IEnumerable<string>>` in [PowerShell Core](https://learn.microsoft.com/en-us/dotnet/api/microsoft.powershell.commands.webresponseobject.headers?view=powershellsdk-7.2.0#microsoft-powershell-commands-webresponseobject-headers)

**Comment by iSazonov** [comment]
As far as I understand header can have multiple values according to RFC.

**Comment by mklement0** [comment]
@iSazonov, maybe it is what you meant, but the issue is not that the header _as a whole_ can have multiple values - by virtue of being implemented as a _dictionary_ that is also true in Windows PowerShell - but whether a single header _entry_ may be composed of multiple strings.

If you have a link to the relevant RFC to share, please do.

**Comment by CarloToso** [comment]
This might be the RFC: https://www.rfc-editor.org/rfc/rfc9110.html section 5. From what I understand PS7 behavour is correct.
Is there something we can do easily about this?
![2](https://user-images.githubusercontent.com/105941898/210670981-1c441f5a-ef8c-4545-9f2b-6e6c15e308fc.png)

**Comment by SteveL-MSFT** [comment]
This definitely looks like an intentional change to support multiple values for a single header

**Comment by iSazonov** [comment]
> but whether a single header _entry_ may be composed of multiple strings.

Yes, exactly.

We migrated from WebClient (Windows PowerShell) to new API - HttpClient (pwsh). As result underlying types were changed too. Now Headers is HttpResponseHeaders type and it is `IEnumerable<KeyValuePair<string, IEnumerable<string>>>`. And although we repackage this to Dictionary, value now remains `IEnumerable<string>`. 
As implementsation detail it is internally `string[]` but I think it is not reliable to use this fact, much less waste resources on repackaging.
